go-controller/pkg/ovn/policy.go

```diff
@@ -1082,9 +1091,9 @@ func (oc *Controller) handlePeerNamespaceAndPodSelector(
 		cache.ResourceEventHandlerFuncs{
 			AddFunc: func(obj interface{}) {
 				namespace := obj.(*kapi.Namespace)
-				np.Lock()
+				np.RLock()
```

Comment:
Do we need this step anymore? Why not just do something like

```
if exists, _ := np.localPods.load(<Pod logical port>); !exists { 
         return
} 
```

Reply:
This is the handler function for namespaces, not pods -- did you mean to comment on a different line?

Reply:
Possibly :) I really just meant why did we need to have the already deleted flag if we can just reference the Localpods sync map....

BUT now I look back and I actually added that bit, it's for the case that the namespace get deleted before a policy delete event is deleted so I guess lets leave it since there's not a great way around that dependency with the current edge driven design